src/components/DiffView/index.tsx

```diff
@@ -254,6 +293,51 @@ export class DiffViewBase extends React.Component<Props> {
       tokens = _tokenize(diff.hunks, options);
     }
 
+    const extraMessages = [];
+
+    let hunks;
+    let diffWasTrimmed = false;
+    let diffIsSlowAlert;
+
+    if (diff) {
+      let changeCount = 0;
+      hunks = diff.hunks.map((hunk) => {
+        changeCount += hunk.changes.length;
+        return hunk;
+      });
+
+      if (changeCount >= _slowDiffChangeCount) {
+        if (!shouldAllowSlowPages(location)) {
+          hunks = trimHunkChanges(hunks, { maxLength: _slowDiffChangeCount });
+          diffWasTrimmed = true;
+        }
+        diffIsSlowAlert = (
+          <SlowPageAlert
+            location={location}
+            getMessage={(allowSlowPages: boolean) => {
```

Comment:
So I guess it would look something more like this?

```jsx
      if (changeCount >= _slowDiffChangeCount) {
        const props = { location };
        diffIsSlowAlert = shouldAllowSlowPages(location) ? (
          <SlowPageAlert
            text={gettext('This diff will load slowly.')}
            linkText={gettext('Shorten the diff.')}
            {...props}
          />
        ) : (
          <SlowPageAlert
            text={gettext('This diff was shortened to load faster.')}
            linkText={gettext('Show the original diff.')}
            {...props}
          />
        );
        extraMessages.push(diffIsSlowAlert);
      }
```

That change doesn't seem like it would help much -- same amount of code, same amount of tests. The `<SlowPageAlert>` component would become simpler, though.


> I assume `SlowPageAlert` was written with the assumption that the parent wouldn't always know what the value of `allowSlowPages` would be, but in practice it seems like any time a component might need to include `SlowPageAlert`

In practice, yes, that's true. It will look the same in `CodeView`: https://github.com/mozilla/addons-code-manager/pull/964/files#diff-1bec0f6c43f589eea9bd31b977d5fa41R101

However, replacing the callbacks with simple text attributes would *require* any calling component to calculate `allowSlowPages` on their own, which feels like an undue burden.

Reply:
An alternative would be to pass 4 props corresponding to the 2 x 2 possible messages.

Reply:
I considered 4 props but 2 props seemed simpler 😄